### PR TITLE
azurecli classic cert auth's service url picked from endpoint

### PR DESCRIPTION
**Pull request by hiyadav**
Azure-CLI classic cert based authentication's Service Management URL will be picked from endpoint.

### CODE REVIEW

File: Tasks/AzureCLI/azureclitask.ts [hunk 1]
```diff
@@ -198,9 +199,9 @@ export class azureclitask {
         }
     }
 
-    private static createPublishSettingFile(subscriptionName:string, subscriptionId:string, certificate:string, publishSettingFileName:string): void  {
+    private static createPublishSettingFile(subscriptionName:string, subscriptionId:string, certificate:string, serviceManagementUrl:string, publishSettingFileName:string): void  {
         //writing the data to the publishsetting file
```
Comment (amaljg) [inline Tasks/AzureCLI/azureclitask.ts]: Just verify if _serviceManagementUrl_ can have a terminal slash and if so, will it cause any issues.

File: Tasks/AzureCLI/azureclitask.ts [hunk 2]
```diff
@@ -130,8 +130,9 @@ export class azureclitask {
         if (endpointAuth.scheme === "Certificate") {
             var bytes = endpointAuth.parameters["certificate"];
             var subscriptionId:string = tl.getEndpointDataParameter(connectedService, "SubscriptionId", true);
+            var serviceManagementUrl:string = tl.getEndpointUrl(connectedService, false);
             const publishSettingFileName:string = path.join(os.tmpdir() ,"subscriptions" + new Date().getTime() + ".publishsettings");
```
Comment (amaljg) [inline Tasks/AzureCLI/azureclitask.ts]: Is the service endpoint URL always present - even for older endpoints which didn't allow selecting this URL? If not, see if we should fallback to the default URL.

File: Tasks/AzureCLI/azureclitask.ts [hunk 3]
```diff
@@ -130,8 +130,9 @@ export class azureclitask {
...
-            this.createPublishSettingFile(subscriptionName, subscriptionId, bytes, publishSettingFileName);
+            this.createPublishSettingFile(subscriptionName, subscriptionId, bytes, serviceManagementUrl, publishSettingFileName);
             var resultOfToolExecution = tl.execSync("azure", "account import \"" + publishSettingFileName + "\"");
```
Comment (amaljg) [inline Tasks/AzureCLI/azureclitask.ts]: UTs for this change?

File: Tasks/AzureCLI/azureclitask.ts (hunk 1 above)
Comment (hiyadav) [inline Tasks/AzureCLI/azureclitask.ts]: while making a call to azure, the azurecli removes the terminal slash. Thus it is not causing any errors.
here is a screenshot of a call to azure while using a publishsetting file with serviceManagementUrl having terminal slash
image

File: Tasks/AzureCLI/azureclitask.ts (hunk 3 above)
Comment (hiyadav) [inline Tasks/AzureCLI/azureclitask.ts]: done!

File: Tasks/AzureCLI/azureclitask.ts (hunk 2 above)
Comment (hiyadav) [inline Tasks/AzureCLI/azureclitask.ts]: a default service URL used to be present in older endpoints. so we should be fine.

**Pull request merged**